authors/apps/profiles/views.py

```diff
@@ -69,3 +75,120 @@ def put(self, request, username):
         else:
 
             return super().put(request, username)
+
+
+class FollowUnfollowAPIView(RetrieveUpdateDestroyAPIView):
+    """
+    APIView for following a user
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, username, format=None):
+        """
+        This method create a user relationship btween the user
+        with the username passed in and the user sending the
+        request
+        """
+        try:
+            to_be_followed = User.objects.get(username=username)
+        except Exception:
+            return Response({
+                'error': "User not found"
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if request.user == to_be_followed:
+            message = {
+                "error": "You cannot follow yourself"
+            }
+            return Response(message, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        try:
+            user = User.objects.get(email=request.user)
+            already_followed = user.following.filter(
+                username=username).exists()
+            if already_followed:
+                return Response({
+                    'error': 'You already follow this user'
+                },
+                    status=status.HTTP_406_NOT_ACCEPTABLE)
+        except Exception as e:
+            return Response({
+                'error': 'Error when following',
+                'message': str(e)
+            },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        request.user.following.add(to_be_followed)
+        request.user.save()
+
+        serializer = FollowUnfollowSerializer(request.user)
+        message = {
+            "message": "Profile successfully followed",
+            "user": serializer.data
+        }
+        return Response(message, status=status.HTTP_200_OK)
+
+    def delete(self, request, username, format=None):
+        """
+        Deletes a follow relationship between the user sending the
+        request and the user with the username passed
+        """
+        try:
+            id = User.objects.get(username=username).id
+        except Exception:
+            return Response({
+                'error': "User not found"
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        follower = User.objects.get(pk=request.user.id)
+
+        relationship = follower.following.filter(pk=id).exists()
+        if not relationship:
+            return Response(
+                {
+                    'error': 'You do not follow this user'
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        follower.following.remove(id)
+        serializer = FollowUnfollowSerializer(follower)
+        message = {
+            "message": "Profile successfully unfollowed",
```

Comment:
As above make the response message descriptive e.g.`You stopped following John Doe`